File: problems/defc7c02-4255-47be-9812-5a9c2468fadd/java/Driver.java
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import java.io.FileReader;
import java.io.IOException;
import java.lang.reflect.Type;
import java.util.List;

public class Driver {
  public static class Testcase {
    public int id;
    public int input;
    public String expected;
    public boolean isPublic;
  }

  public static List<Testcase> loadTestcases(String filename) {
    try (FileReader reader = new FileReader(filename)) {
      Gson gson = new Gson();
      Type testcaseListType = new TypeToken<List<Testcase>>(){}.getType();
      return gson.fromJson(reader, testcaseListType);
    } catch (IOException e) {
      System.err.println("Failed to open testcases file.");
      System.exit(1);
      return null;
    }
  }
}

File: problems/defc7c02-4255-47be-9812-5a9c2468fadd/cpp/driver.cpp
using json = nlohmann::json;
using namespace std;

namespace Driver {
    struct Testcase {
        int id;
        int input;
        string expected;
        bool isPublic;
    };

    vector<Testcase> loadTestcases(const string& filename) {
        ifstream file(filename);
        if (!file.is_open()) {
            cerr << "Failed to open testcases file." << endl;
            exit(1);
        }

        json j;
        file >> j;

        vector<Testcase> testcases;
        for (const auto& item : j) {
            Testcase tc;
            tc.id = item["id"];
            tc.input = item["input"];
            tc.expected = item["expected"];
            tc.isPublic = item["isPublic"];
            testcases.push_back(tc);
        }

        return testcases;
    }
}
